lib/vrl/stdlib/src/to_regex.rs

```diff
@@ -0,0 +1,76 @@
+use regex::Regex;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToRegex;
+
+impl Function for ToRegex {
+    fn identifier(&self) -> &'static str {
+        "to_regex"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::bytes,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "regex",
+                source: "to_regex(s'foo')",
+                result: Ok(r"foo"),
+            }
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        Ok(Box::new(ToRegexFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ToRegexFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ToRegexFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        use Value::*;
+
+        let value = match self.value.resolve(ctx)? {
+            v @ Bytes(_) => Value::Regex::new(v).unwrap(),
+            v => return Err(format!(r#"unable to coerce {} into "regex""#, v.kind()).into()),
+        };
```

Comment:
You can just use `let value = self.value.resolve(ctx)?.try_bytes()?` here instead of matching and return a custom error.

Reply:
But isn't using `try_bytes()?` make the produced value still a `string`, instead of a `regex`?

Reply:
Ah, sorry, I should have been more clear. What you ended up with is close to what I was thinking.